python/cudf/cudf/core/udf/utils.py

```diff
@@ -420,3 +441,30 @@ def maybe_patch_numba_linker(
             Linker.new = new_patched_linker
         else:
             logger.debug("Cannot patch Numba Linker - unsupported version")
+
+
+@initfunc
+def set_malloc_heap_size(size=None):
+    """
+    Heap size control for strings_udf, size in bytes.
+    """
+    global _heap_size
+    if size is None:
+        size = _STRINGS_UDF_DEFAULT_HEAP_SIZE
+    if size != _heap_size:
+        (ret,) = cudart.cudaDeviceSetLimit(
+            cudart.cudaLimit.cudaLimitMallocHeapSize, size
+        )
+        if ret.value != 0:
+            raise RuntimeError("Unable to set cudaMalloc heap size")
+
+        _heap_size = size
+
+
+def column_to_string_view_array_init_heap(col):
+    # lazily allocate heap only when a string needs to be returned
+    return column_to_string_view_array(col)
+
+
+strings_ptx_file = _get_ptx_file(os.path.dirname(__file__), "shim_")
+ptx_files.append(strings_ptx_file)
```

Comment:
Why don't we just initialize `ptx_files` with this value above? We don't need to initialize-and-modify (append).

Reply:
I refactored this a bit in https://github.com/rapidsai/cudf/pull/12669/commits/c304f5c75165ae1528066765b2b73d6e45a14da5 .